src/main/resources/db/migration/V2_29__add_event_audit.sql

```diff
@@ -0,0 +1,49 @@
+CREATE TABLE event_audit
+(
+    id                          serial          NOT NULL PRIMARY KEY,
+    booking_reference           text,
+    application_reference       text,
+    session_template_reference  text,
+    type                        varchar(90) NOT NULL,
+    application_method_type     varchar(90) NOT NULL,
+    actioned_by                 varchar(60),
+    create_timestamp            timestamp default current_timestamp
+);
+
+CREATE INDEX idx_booking_reference ON event_audit(booking_reference);
+CREATE INDEX idx_visit_application_reference ON event_audit(application_reference);
+CREATE INDEX idx_visit_session_template_reference ON event_audit(session_template_reference);
+
+-- Copy data over from session_template table
+INSERT INTO event_audit (booking_reference, application_reference, session_template_reference, type, application_method_type, actioned_by, create_timestamp)
+    (SELECT reference,
+            application_reference,
+            session_template_reference,
+            CASE
+                WHEN visit_status   = 'RESERVED' THEN 'RESERVED_VISIT'
+                WHEN visit_status   = 'CHANGING' THEN 'CHANGING_VISIT'
+                WHEN visit_status   = 'BOOKED' THEN 'BOOKED_VISIT'
+                WHEN outcome_status = 'SUPERSEDED_CANCELLATION' THEN 'UPDATED_VISIT'
```

Comment:
will it have UPDATED_VISIT on the first entry?

Reply:
don't think so,.. will ether be RESERVED_VISIT or CHANGING_VISIT

Reply:
this is the code that needs most attention

Reply:
I mean if the visit is Updated will it end up with UPDATED_VISIT as the first entry?